src/main/java/seedu/address/storage/JsonTranscriptDeserializer.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.storage;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.ReadOnlyTranscript;
+import seedu.address.model.Transcript;
+import seedu.address.model.module.Module;
+
+/**
+ * Deserializer for {@link seedu.address.model.ReadOnlyTranscript}.
+ */
+public class JsonTranscriptDeserializer extends StdDeserializer<ReadOnlyTranscript> {
+    
+
+    private static final Logger logger = LogsCenter.getLogger(JsonTranscriptStorage.class);
+    public static final String MESSAGE_DUPLICATE_MODULE = "Transcript in JSON file contains duplicate modules.";
+    public static final String MESSAGE_INVALID_TRANSCRIPT = "Transcript in JSON contains invalid data:";
+
+    public JsonTranscriptDeserializer(Class<ReadOnlyTranscript> vc) {
+        super(vc);
+    }
+
+    public JsonTranscriptDeserializer() {
+        super(ReadOnlyTranscript.class);
+    }
+    
+    @Override
+    public ReadOnlyTranscript deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+        Transcript transcript = new Transcript();
+        JsonNode node = jp.getCodec().readTree(jp);
+        try {
+            Iterator<JsonNode> elements = node.get("modules").get("internalList").elements();
+            for (; elements.hasNext(); ) {
```

Comment:
Might be a readability issue, maybe a while loop would be more intuitive? But yea otherwise it works.

Reply:
Thanks for the quick review mate! Yeap sorry I struggled quite abit with this whole part, this bit I copied from Stack Overflow haha and forgot to change. Will update.